packages/components/organisms/f-form/src/components/Form.vue

```diff
@@ -1,46 +1,102 @@
 <template>
-    <div
+    <form
+        method="post"
         :class="$style['c-form']"
-        data-test-id="form-component">
-        {{ copy.text }}
-    </div>
+        data-test-id="form-component"
+        @submit.prevent="onFormSubmit">
+        <form-field
+            v-for="field in formData.formFields"
+            :key="`${field.name}-field`"
+            v-bind="fieldProps(field)"
+            @input="updateField({ fieldName: field.name, value: $event })" />
+
+        <f-button
+            button-type="primary"
+            button-size="large"
+            :class="$style['c-form-submitButton']"
+            is-full-width
+            action-type="submit"
+            data-test-id="submit-button"
+            :is-loading="isFormSubmitting">
+            {{ formData.buttonText }}
+        </f-button>
+    </form>
 </template>
 
+
 <script>
+import FButton from '@justeat/f-button';
+import FormField from '@justeat/f-form-field';
 import { globalisationServices } from '@justeat/f-services';
 import tenantConfigs from '../tenants';
 
 export default {
-    name: 'VForm',
-    components: {},
+    components: {
+        FButton,
+        FormField
+    },
+
     props: {
-        locale: {
-            type: String,
-            default: ''
+        formData: {
+            type: Object,
+            required: true
+        },
+
+        isFormSubmitting: {
+            type: Boolean,
+            default: false
         }
     },
+
     data () {
         const locale = globalisationServices.getLocale(tenantConfigs, this.locale, this.$i18n);
         const localeConfig = tenantConfigs[locale];
 
         return {
             copy: { ...localeConfig }
         };
+    },
+
+    computed: {
+        formFields () {
+            const formFields = {};
+
+            this.formData.formFields.forEach(field => {
+                formFields[field.name] = field.value;
+            });
+
+            return formFields;
+        }
+    },
+
+    methods: {
+        updateField ({ fieldName, value }) {
+            return this.$emit('updated', { fieldName, value });
+        },
+
+        onFormSubmit () {
+            this.$emit('form-submitting');
+        },
+
+        fieldProps (field) {
+            return {
+                name: field.name,
+                value: field.value,
```

Comment:
You could make this default to empty and then if the `field.value` is set then use that value

```suggestion
                value: field.value || '',
```

That would save having to pass in an empty string for field objects with no default value.